Chrysalis Docs Review IV (#72)
iotaledger/chrysalis-docs: docs/faq.md

# FAQ

## Which Libraries are Available for Chrysalis?

At this time, there are a handful of libraries available for Chrysalis which are outlined below:

> 

Reviewer: **For line 8**
- Please add a link to the iota.rs repository or documentation.

**For line 10**
- Please add a link to the wallet.rs repository or documentation.
- Is it coins or tokens?

Author: Coins, I believe!

Additionally, there is also a native C ([iota.c](https://github.com/iotaledger/iota.c)) and an alternative, native javascript ([iota.js](https://github.com/iotaledger/iota.js/tree/chrysalis)) client library. 

## What is Dust Protection and How Does it Work?

Since IOTA is feeless and has the ability to send microtransactions, attackers could use this to spam the network with very low value transactions, which we call dust. To avoid this, we only allow microtransaction below 1Mi of IOTA tokens to another address if you already have at least 1Mi on that address.

> You can read more about Dust Protection [in the RFC here](https://github.com/iotaledger/protocol-rfcs/pull/32).

Reviewer: - The provided link is for an open pull request, perhaps we should change this to a blog post or some documentation which explains this better.
- Instead of saying 'here', we could say 'this pull request' (or blog post/documentation if we change the link)

Author: So, we actually reference the RFCs a lot as its assumed the audience wants to go much around in our stuff. I agree that there _should_ be a reference to more digestible content but it doesnt exist at the moment :/

## What is Happening with Coordicide?

> This release called Chrysalis still depends on the Coordinator/Compass to run and issue milestones. The next big release after Chrysalis will be Coordicide where we get rid of that dependency. This release is currently in a research/testing phase and will be the main priority after the Chrysalis release.

Reviewer: ```suggestion
The Chrysalis release still depends on the Coordinator/Compass to run and issue milestones. The next big release after Chrysalis will be Coordicide, where we will get rid of that dependency. This release is currently in a research/testing phase, and will be the main priority after the Chrysalis release.

## When is a Transaction on the Network Considered Final/Irreversible?

> Confirmation times on the new network are on average around 10 seconds. Once a transaction is set to confirm it’s final transaction, you don’t have block confirmations like with blockchain.

Reviewer: - I'd change the first sentence to:
'On average, confirmation times on the new network are around 10 seconds. '
- Perhaps it's just me as I'm kind of new to this, but I don't understand the second sentence.

Author: The "confirmation" is a number of times another block or transaction is placed chronologically after the transaction's block. So its not "block like a goalie" so much as "lego blocks (bricks but you get the idea) stacking after your transaction. So _your_ block is only necessary to confirm, there is no waiting for _other_ blocks. Sorry if that wasn't explained clearly.

## Hornet or Bee? Which Node Software Should I Use?

> You can either pick Bee (Rust based) or Hornet (Go based). We currently recommend Hornet since it’s the more complete version of the node software, Bee is still missing some optional features you might wish to use.

Reviewer: ```suggestion
You can either pick [Bee](https://bee.docs.iota.org/) (Rust based) or [Hornet](https://hornet.docs.iota.org/) (Go based). We currently recommend Hornet, since it’s the most complete version of the node software.  Bee is still missing some optional features you might wish to use.
```

- We could also list a couple of the missing optional features here so the user can choose if he needs them or not.

Author: I _think_ the original authors intentionally left the optional features blank. Looking at the language around them "although this is probably not useful for IOTA's use case", I'd side with the authors on this one.

## What are Spent Addresses and Why are they Dangerous?
- In the IOTA 1.0 network, IOTA used Winternitz One Time Signatures (W-OTS) - think of these as, more or less, an authenticator and validator for a transaction. These keys and signatures are highly secure against malicious attacks for signing transactions. But, on the downside, by signing a transaction, W-OTS reveals parts of a private key for the specific address tokens are being spent from. 

> - With W-OTS every time a signature is signed to spend tokens from a particular address, any remaining tokens need to be moved onto a new address to prevent malicious actors from brute-forcing (trial-and-error guessing) the remaining parts of the private key for the address. That’s the main reason why this signature scheme is considered to be a “one-time signature”. 

Reviewer: - "a signature is signed" seems a bit redundant.

Author: Updated!

> - So after our Chrysalis update, we are now using the Ed25519, based on the EdDSA, scheme instead of W-OTS. The advantage is that the new scheme addresses all of the issues that W-OTS originally had, where Ed25519 verifies both single-signature and batch verification (taking care of the left over/remaining tokens) very quickly as well as faster key generation and smaller signatures (very secure). 

Reviewer: ```suggestion
After our Chrysalis update, we are now using the Ed25519, based on the EdDSA scheme instead of W-OTS. The advantage is that the new scheme addresses all the issues that W-OTS originally had. Ed25519 verifies both single-signature and batch verification (taking care of the left over/remaining tokens) very quickly, as well as faster key generation and smaller signatures which make it very secure.
```

Author: I _liked_ my parentheses...but updated!

## What is Bundle Mining?

> -  If you have spent addresses, it means you accidentally received funds to an address that was already spent from and these funds are not safe to send again due to W-OTS. 

Reviewer: ```suggestion
If you have spent addresses, it means you accidentally received funds to an address that was already spent from, and these funds are not safe to send again due to W-OTS.
```

> -  To secure your spent addresses during the migration, Firefly will try to find a new bundle that reveals the least amount of additional private key parts compared to previous signs.

Reviewer: - We could add a link to the firelfy repo or docs.
- Is it the previous signs or signatures?

> - This process will take 10 minutes per spent address and upon completion you will be presented with a risk calculation (very high, high, medium, low, very low). It is recommended that you repeats the process if it returns a bundle with medium risk or higher, particularly for significant sums of IOTA. You have the option to select which addresses you want to mine for and again which you want to rerun the process for.

Reviewer: ```suggestion
This process will take 10 minutes per spent address and upon completion you will be presented with a risk calculation (very high, high, medium, low, very low).  We recommend that you repeat the process if it returns a bundle with medium risk or higher, particularly for significant sums of IOTA. You have the option to select which addresses you want to mine for, and which you want to rerun the process for.
```
